common/persistence/persistenceTestBase.go

```diff
@@ -106,66 +109,102 @@ func newTestShardContext(shardInfo *ShardInfo, transferSequenceNumber int64, his
 	}
 }
 
-func (s *testShardContext) GetExecutionManager() ExecutionManager {
+// GetExecutionManager test implementation
+func (s *TestShardContext) GetExecutionManager() ExecutionManager {
 	return s.executionMgr
 }
 
-func (s *testShardContext) GetHistoryManager() HistoryManager {
+// GetHistoryManager test implementation
+func (s *TestShardContext) GetHistoryManager() HistoryManager {
 	return s.historyMgr
 }
 
-func (s *testShardContext) GetNextTransferTaskID() (int64, error) {
+// GetNextTransferTaskID test implementation
+func (s *TestShardContext) GetNextTransferTaskID() (int64, error) {
 	return atomic.AddInt64(&s.transferSequenceNumber, 1), nil
 }
 
-func (s *testShardContext) GetTransferMaxReadLevel() int64 {
+// GetTransferMaxReadLevel test implementation
+func (s *TestShardContext) GetTransferMaxReadLevel() int64 {
 	return atomic.LoadInt64(&s.transferSequenceNumber)
 }
 
-func (s *testShardContext) GetTransferAckLevel() int64 {
+// GetTransferAckLevel test implementation
+func (s *TestShardContext) GetTransferAckLevel() int64 {
 	return atomic.LoadInt64(&s.shardInfo.TransferAckLevel)
 }
 
-func (s *testShardContext) GetTimerSequenceNumber() int64 {
-	return atomic.AddInt64(&s.timerSequeceNumber, 1)
-}
-
-func (s *testShardContext) UpdateAckLevel(ackLevel int64) error {
+// UpdateTransferAckLevel test implementation
+func (s *TestShardContext) UpdateTransferAckLevel(ackLevel int64) error {
 	atomic.StoreInt64(&s.shardInfo.TransferAckLevel, ackLevel)
 	return nil
 }
 
-func (s *testShardContext) GetTransferSequenceNumber() int64 {
+// GetTransferSequenceNumber test implementation
+func (s *TestShardContext) GetTransferSequenceNumber() int64 {
 	return atomic.LoadInt64(&s.transferSequenceNumber)
 }
 
-func (s *testShardContext) CreateWorkflowExecution(request *CreateWorkflowExecutionRequest) (
+// GetTimerAckLevel test implementation
+func (s *TestShardContext) GetTimerAckLevel() time.Time {
+	s.Lock()
```

Comment:
The type is a timestamp, there is no direct type on atom.Load*, i thought of using atomic.StorePointer, but then this is test code and thought of not using pointer swap. 
i didn't use RWMutex thinking this is a test code and probably doesn't make much difference on a second thought going to replaces with RW mutex.